crux-mir/src/Mir/Language.hs

```diff
@@ -571,6 +573,11 @@ mirConfig = Crux.Config
         , GetOpt.Option []  ["cargo-test-file"]
             "treat trailing args as --test-filter instead of FILES, like `cargo test`; load program from this file instead (used by `cargo crux test`)"
             (GetOpt.ReqArg "file" (\v opts -> Right opts { cargoTestFile = Just v }))
+
+        , GetOpt.Option [] ["mir-json-arg"]
+            "pass an argument to mir-json (can be repeated)"
+            (GetOpt.ReqArg "ARG" $ \v opts ->
+                Right opts { mirJsonArgs = mirJsonArgs opts ++ [v] })
```

Comment:
This appends new entries to the end of the list each time, and since `(++)` is an O(n) operation, it will take O(n^2) time to process all of the `mir-json-arg`s. I think it would be worth using a different data structure with a cheaper append operation (e.g., [`Data.Sequence`](https://hackage-content.haskell.org/package/containers-0.8/docs/Data-Sequence.html#t:Seq)) instead of a list to avoid this problem.